compiler-plugin/src/main/java/io/ballerina/stdlib/constraint/compiler/annotation/tag/AnnotationTagFloat.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.constraint.compiler.annotation.tag;
+
+import java.util.ArrayList;
+
+import static io.ballerina.stdlib.constraint.compiler.Constants.CONSTRAINT_MAX_VALUE;
+import static io.ballerina.stdlib.constraint.compiler.Constants.CONSTRAINT_MAX_VALUE_EXCLUSIVE;
+import static io.ballerina.stdlib.constraint.compiler.Constants.CONSTRAINT_MIN_VALUE;
+import static io.ballerina.stdlib.constraint.compiler.Constants.CONSTRAINT_MIN_VALUE_EXCLUSIVE;
+import static io.ballerina.stdlib.constraint.compiler.Constants.TYPE_FLOAT;
+
+/**
+ * The class to represent the `@constraint:Float` annotation tag.
+ */
+public class AnnotationTagFloat implements AnnotationTag {
+
+    @Override
+    public boolean isCompatibleFieldType(String fieldType) {
+        return fieldType.equals(TYPE_FLOAT);
+    }
+
+    @Override
+    public boolean haveCompatibleConstraints(ArrayList<String> constraints) {
```

Comment:
This method seems duplicated among other derived classes. Can we improve the abstraction?

Reply:
Fixed with another level of abstraction with https://github.com/ballerina-platform/module-ballerina-constraint/pull/1/commits/27f6a1820945a94165eb0906f558c41e1401b1a1